packages/teleport-plugin-common/src/utils/ast-utils.ts

```diff
@@ -610,19 +610,33 @@ export const generateRemoteResourceASTs = (resource: UIDLResourceItem) => {
       )
     : fetchUrl
 
+  const method = types.objectProperty(
+    types.identifier('method'),
+    types.stringLiteral(resource.method)
+  )
+
+  let allHeaders: types.ObjectProperty[] = []
+
+  if (authHeaderAST) {
+    allHeaders.push(authHeaderAST)
+  }
+
+  if (headersASTs.length) {
+    allHeaders = allHeaders.concat(headersASTs)
+  }
+
+  const headers = types.objectProperty(
+    types.identifier('headers'),
+    types.objectExpression(allHeaders.length ? [...allHeaders] : [])
```

Comment:
You can directly use it this way

```js
types.objectExpression(allHeaders)
```

as it is being declared as empty array anyways